source/backend/api/Services/DocumentFileService.cs

```diff
@@ -135,6 +170,24 @@ public async Task<ExternalResult<string>> DeleteResearchDocumentAsync(PimsResear
             }
         }
 
+        public async Task<ExternalResult<string>> DeleteProjectDocumentAsync(PimsProjectDocument projectDocument)
+        {
+            this.Logger.LogInformation("Deleting PIMS document for single Project");
+            this.User.ThrowIfNotAuthorized(Permissions.DocumentDelete);
+
+            IList<PimsProjectDocument> existingProjectDocuments = _projectRepository.GetAllByDocument(projectDocument.DocumentId);
+            if (existingProjectDocuments.Count == 1)
+            {
+                return await documentService.DeleteDocumentAsync(projectDocument.Document);
+            }
```

Comment:
This does not look correct. I think it should only delete the document if there are no other associations to the document, including other files, leases, etc. As it is, this one only checks if the document has a project document.

Reply:
Updated to check the total number of relationships instead of just one type.